### PR TITLE
feat: data error handling

### PR DESCRIPTION
**Pull request by kartikCypherock**
Implements https://app.clickup.com/t/85zt3zq2f

### CODE REVIEW

**Comment by amanCypherock** [comment]
Can we break this PR such that the manager APIs are merged in 1 PR and other app functions handling are taken up 1 PR at a time?
Currently, this PR is blocked as we have to review, and test every function of the manager app and would need to do follow-ups accordingly. If we merge the APIs first, we will have an option to use these functions and merge of one flow will not block the other.

**Comment by kartikCypherock** [comment]
> Can we break this PR such that the manager APIs are merged in 1 PR and other app functions handling are taken up 1 PR at a time? Currently, this PR is blocked as we have to review, and test every function of the manager app and would need to do follow-ups accordingly. If we merge the APIs first, we will have an option to use these functions and merge of one flow will not block the other.

I have reverted change from `auth_card.c`, for other files, we have already integrated new APIs, so it is not feasible to revert the change. Please re-review

File: apps/manager_app/manager_api.c [hunk 1]
```diff
@@ -72,6 +75,13 @@
 /*****************************************************************************
  * PRIVATE MACROS AND DEFINES
  *****************************************************************************/
+#define P0_SESSION_HANDLER(event)                                              \
+  do {                                                                         \
+    if (true == event.p0_event.flag) {                                         \
+      manager_send_data_flow_error(ERROR_DATA_FLOW_INACTIVITY_TIMEOUT);        \
```
Comment (ujjwal-cyph) [inline apps/manager_app/manager_api.c:81]: abort event should not send this error

Comment (kartikCypherock) [reply apps/manager_app/manager_api.c]: fixed

File: apps/manager_app/joystick_training.c [hunk 2]
```diff
@@ -218,7 +180,15 @@ static bool training_step(const joystick_step_t *step) {
 void manager_joystick_training(manager_query_t *query) {
   /* Validate if this flow is allowed */
   if (false == onboarding_step_allowed(ONBOARDING_JOYSTICK_TRAINING)) {
-    // TODO: Reject query
+    manager_send_data_flow_error(ERROR_DATA_FLOW_INVALID_REQUEST);
+    return;
+  }
+
+  /* Validate request */
+  if (MANAGER_TRAIN_JOYSTICK_REQUEST_INITIATE_TAG ==
```
Comment (ujjwal-cyph) [inline apps/manager_app/joystick_training.c:188]: this check should be inverted `!=` in place of `==`

File: apps/manager_app/manager_api.c [hunk 3]
```diff
@@ -72,6 +75,15 @@
 /*****************************************************************************
  * PRIVATE MACROS AND DEFINES
  *****************************************************************************/
+#define P0_SESSION_HANDLER(event)                                              \
+  do {                                                                         \
+    if (true == event.p0_event.flag) {                                         \
+      if (true == event.p0_event.inactivity_evt) {                             \
+        manager_send_data_flow_error(ERROR_DATA_FLOW_INACTIVITY_TIMEOUT);      \
```
[the remark this answers is not in the document]
Comment (kartikCypherock) [reply apps/manager_app/manager_api.c:82]: updated

**Review by ujjwal-cyph — APPROVED** [review]

**Comment by amanCypherock** [comment]
Could you update the submodule here? The latest cypherock-common submodule commit is 95028550f962e97302aa454dab822a1bc721d17e

**Comment by kartikCypherock** [comment]
> Could you update the submodule here? The latest cypherock-common submodule commit is 95028550f962e97302aa454dab822a1bc721d17e

Ok, done

**Review by amanCypherock — APPROVED** [review]

**Pull request merged**